port/unix/omrmmap.c

```diff
@@ -50,11 +49,98 @@
 #include "protect_helpers.h"
 #include "omrportpriv.h"
 
+#if defined(J9ZOS390)
+#include "omrvmem.h"
+#if !defined(MAP_FAILED)
+#define MAP_FAILED ((void*)-1)
```

Comment:
This cast should map to `intptr_t` first to ensure all bits are set:
```c
#define MAP_FAILED ((void *)(intptr_t)-1)
```

Reply:
Fixed.